server/engine/src/main/java/org/eclipse/lsp/cobol/core/visitor/CobolVisitor.java

```diff
@@ -190,6 +197,7 @@ public List<Node> visitWorkingStorageSection(WorkingStorageSectionContext ctx) {
   @Override
   public List<Node> visitProgramUnit(ProgramUnitContext ctx) {
     fileControls = new HashMap<>();
+    text.reset();
```

Comment:
does this work correctly with the nested programUnits?

Reply:
in some sense
![image](https://github.com/user-attachments/assets/c12dc3d2-8a42-4538-9cce-c189097bf519)

Reply:
I am wondering if the `text` should really be a stack as well.